samcli/commands/init/init_templates.py

```diff
@@ -112,14 +106,23 @@ def _check_app_template(entry: Dict, app_template: str) -> bool:
         return bool(entry["appTemplate"] == app_template)
 
     def init_options(self, package_type, runtime, base_image, dependency_manager):
-        if not self.clone_attempted:
-            self._clone_repo()
-        if self.repo_path is None:
+        if not self._git_repo.clone_attempted:
+            shared_dir: Path = global_cfg.config_dir
+            try:
+                self._git_repo.clone(clone_dir=shared_dir, replace_existing=True)
+            except CloneRepoUnstableStateException as ex:
+                raise UserException(str(ex)) from ex
+            except (OSError, CloneRepoException):
+                # If can't clone, try using an old clone from a previous run if already exist
+                expected_previous_clone_local_path: Path = shared_dir.joinpath(self._git_repo.name)
+                if expected_previous_clone_local_path.exists():
+                    self._git_repo.local_path = expected_previous_clone_local_path
```

Comment:
Where is the corresponding logic in the old code?

Reply:
https://github.com/aws/aws-sam-cli/blob/develop/samcli/commands/init/init_templates.py#L183